File: on-boarding-samples/on-boarding-sample-java/src/main/java/apio/architect/example/PersonActionRouter.java
package apio.architect.example;

import com.liferay.apio.architect.annotation.Actions.Retrieve;
import com.liferay.apio.architect.annotation.EntryPoint;
import com.liferay.apio.architect.annotation.Id;
import com.liferay.apio.architect.router.ActionRouter;

import org.osgi.service.component.annotations.Component;

import java.util.Arrays;
import java.util.List;


@Component
public class PersonActionRouter implements ActionRouter<Person> {

    @EntryPoint
    @Retrieve
    public List<Person> getPersons() {
        return Arrays.asList(
            Person.of(1, "Alex", "Developer"),
            Person.of(2, "David", "Developer"));
    }

    @Retrieve
    public Person getPerson(@Id long personId) {
        return Person.of(1, "Alex", "Developer");
    }

}

File: on-boarding-samples/on-boarding-sample-kotlin/src/main/java/apio/architect/example/PersonActionRouter.kt
package apio.architect.example

import com.liferay.apio.architect.router.ActionRouter
import com.liferay.apio.architect.annotation.Actions.Retrieve
import com.liferay.apio.architect.annotation.EntryPoint
import com.liferay.apio.architect.annotation.Id

import org.osgi.service.component.annotations.Component

@Component
class PersonActionRouter : ActionRouter<Person> {

    @EntryPoint @Retrieve fun getPersons() =
            listOf(Person.of(1, "Alex", "Developer"), Person.of(2, "David", "Developer"))

    @Retrieve fun getPerson(@Id personId: Long) = Person.of(1, "Alex", "Developer")

}